src/main/java/de/retest/recheck/review/ignore/matcher/ElementByAttributeMatcher.java

```diff
@@ -0,0 +1,50 @@
+package de.retest.recheck.review.ignore.matcher;
+
+import java.util.Optional;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+
+import de.retest.recheck.review.ignore.io.RegexLoader;
+import de.retest.recheck.ui.descriptors.Element;
+
+public class ElementByAttributeMatcher implements Matcher<Element> {
+
+	private final String key;
+	private final String value;
+
+	public ElementByAttributeMatcher( final String key, final String value ) {
+		this.key = key;
+		this.value = value;
+	}
+
+	@Override
+	public boolean test( final Element element ) {
+		Object elementAttribute = element.getIdentifyingAttributes().get( key );
+		if ( elementAttribute != null ) {
+			return value.equals( elementAttribute );
+		}
+		elementAttribute = element.getAttributes().get( key );
+		return value.equals( elementAttribute );
```

Comment:
Just a suggestion to use different names for the local variables as well as Optional for clarity.
```suggestion
		final Optional<Object> elementIdentifyingAttribute =
				Optional.ofNullable( element.getIdentifyingAttributes().get( key ) );
		if ( elementIdentifyingAttribute.isPresent() ) {
			return value.equals( elementIdentifyingAttribute.get() );
		}
		final Object elementAttributes = element.getAttributes().get( key );
		return value.equals( elementAttributes );
```

Reply:
I feel that this makes the code harder to read...

Reply:
Hard to discuss 'feelings'.

Ideally `element.getIdentifyingAttributes().get( key )` should return an Optional<Object> (or even better, an Optional<someInterface>), not null or Object.
Personally, I prefer to use `isPresent()` instead of `!= null`. 

But of course this is your decision and this will not be taken into account for the PR approval.